orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/kubernetes/KubernetesServerGroupCreator.groovy

```diff
@@ -48,8 +48,7 @@ class KubernetesServerGroupCreator implements ServerGroupCreator {
     containers.forEach { container ->
       if (container.imageDescription.fromContext) {
         def image = deploymentDetails.find {
-          def cluster = Names.parseName(it.sourceServerGroup).cluster
-          cluster == container.imageDescription.cluster && it.imageName ==~ container.imageDescription.pattern
+          it.refId == container.imageDescription.stageId
```

Comment:
Might be worth dropping a comment here to explain why you're using the `refId (basically the text from the PR description).